example/lib/ios_popover/_ios_toolbar.dart

```diff
@@ -0,0 +1,594 @@
+    }
+
+    // Page to be displayed.
+    final currentPage = _pages![_controller.currentPage - 1];
+
+    if (hasMultiplePages) {
+      // Computes previous button position.
+      final previousButton = children.first;
+      final previousButtonParentData = previousButton.parentData as _IosPagerParentData;
+      previousButtonParentData.offset = Offset(width, (height - previousButton.size.height) / 2);
+
+      // Update current width.
+      width += previousButton.size.width;
+    }
+
+    // Set offset of children which belong to current page.
+    for (int i = currentPage.startingIndex; i < currentPage.endingIndex; i++) {
+      final child = children[i];
+      final childSize = child.size;
+      final childParentData = child.parentData as _IosPagerParentData;
+      childParentData.offset = Offset(width, (height - childSize.height) / 2);
+
+      // Update current width.
+      width += childSize.width;
+    }
+
+    if (hasMultiplePages) {
+      // Computes next button position.
+      final nextButton = children.last;
+      final nextButtonButtonParentData = nextButton.parentData as _IosPagerParentData;
+      nextButtonButtonParentData.offset = Offset(width, (height - nextButton.size.height) / 2);
+
+      // Update current width.
+      width += nextButton.size.width;
+    }
+
+    size = Size(width, height);
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    final children = getChildrenAsList();
+    final page = _pages![_controller.currentPage - 1];
+
+    late Offset childOffset;
+    final hasMultiplePages = (_pages?.length ?? 0) > 1;
+
+    if (hasMultiplePages) {
+      // Paint the previous page button.
+      final previousButton = children.first;
+      childOffset = (previousButton.parentData as _IosPagerParentData).offset;
+      context.paintChild(previousButton, childOffset + offset);
+    }
+
+    for (int i = page.startingIndex; i < page.endingIndex; i++) {
+      final child = children[i];
+      childOffset = (child.parentData as _IosPagerParentData).offset;
+
+      if (hasMultiplePages || i > page.startingIndex) {
+        // Paint the separator.
+        context.canvas.drawLine(
+          offset + Offset(childOffset.dx, 0),
+          offset + Offset(childOffset.dx, size.height),
+          _linePaint,
+        );
+      }
+
+      // Paint the child content.
+      context.paintChild(child, childOffset + offset);
+    }
+
+    if (hasMultiplePages) {
+      final nextButton = children.last;
+      childOffset = (nextButton.parentData as _IosPagerParentData).offset;
+
+      // Paint the separator.
+      context.canvas.drawLine(
+        offset + Offset(childOffset.dx, 0),
+        offset + Offset(childOffset.dx, size.height),
+        _linePaint,
+      );
+
+      // Paint the next page button.
+      context.paintChild(nextButton, childOffset + offset);
+    }
+  }
+
+  @override
+  bool hitTestChildren(BoxHitTestResult result, {required Offset position}) {
```

Comment:
What's different about the hit testing that we need here vs. the default hit testing implementation?

Reply:
We only hit test the items of the current page.